src/app/TheoIDE/Controls/include/editormodel.hpp

```diff
@@ -165,32 +200,48 @@ class EditorModel : public QAbstractListModel {
                                                 const FileReadError& error);
   void displayFileReadFileDoesNotExistFailure(
       QSharedPointer<QFile> file, const FileDoesNotExistError& error);
+  void stopExecution();
+  void setExecutionTimeoutMs(int value);
+  void setCompilationTimeoutMs(int value);
 
  signals:
   void currentTabIndexChanged(int index);
   void compilerServiceChanged();
   void dialogServiceChanged();
   void fileSystemServiceChanged();
+  void virtualMachineServiceChanged();
   void mainTabIndexChanged(int index);
-  void isRunningChanged(bool isRunning);
+  void runningModeChanged(RunningMode runningMode);
+  void executionTimeoutMsChanged();
+  void compilationTimeoutMsChanged();
+  void executionStateChanged();
 
  protected slots:
   void updateMainTabIndex();
   void updateAllTabNames();
-  void compilationRevisionAvailable(int revision);
+  void handleCompilationRevisionAvailable(int revision);
+  void handleExecutionCompleted();
+  void displayExecutionFailedForInternalReason();
+  void updateRevision();
```

Comment:
This method is not implemented in the .cpp file and causes a linking error "undefined reference to updateRevision()".
When I comment out this line, the build succeeds. Maybe this method was planned but isn't needed? Probably happens because I build statically on my system by default.